nexus-sdk-js/src/Resource/index.ts

```diff
@@ -0,0 +1,73 @@
+import { Context, Distribution } from '../utils/types';
+import {
+  ResourceMetadataKeys,
+  ResourceResponseCommon,
+  ResourceResponse,
+  ListResourceResponse,
+} from './types';
+
+export {
+  ResourceResponse,
+  ResourceResponseCommon,
+  ResourceMetadataKeys,
+  ListResourceResponse,
+};
+
+export default class Resource<T = {}> {
+  orgLabel: string;
+  projectLabel: string;
+  context?: Context;
+  type?: string[];
+  self: URL;
+  id: string;
+  constrainedBy: string;
+  project: string;
+  createdAt: string;
+  createdBy: string;
+  updatedAt: string;
+  updatedBy: string;
+  rev: number;
+  deprecated: boolean;
+  distribution?: Distribution | Distribution[];
+  data: T;
+  constructor(
+    orgLabel: string,
+    projectLabel: string,
+    resourceResponse: ResourceResponseCommon,
+  ) {
+    this.orgLabel = orgLabel;
+    this.projectLabel = projectLabel;
+    this.id = resourceResponse['@id'];
+    if (resourceResponse['@type']) {
+      if (Array.isArray(resourceResponse['@type'])) {
+        this.type = resourceResponse['@type'] as Array<string>;
+      } else {
+        this.type = new Array(resourceResponse['@type']) as Array<string>;
+      }
+    }
+    if (resourceResponse['@context']) {
+      this.context = resourceResponse['@context'];
+    }
+    this.self = new URL(resourceResponse._self);
+    this.constrainedBy = resourceResponse._constrainedBy;
+    this.project = resourceResponse._project;
+    this.createdAt = resourceResponse._createdAt;
+    this.createdBy = resourceResponse._createdBy;
+    this.updatedAt = resourceResponse._updatedAt;
+    this.updatedBy = resourceResponse._updatedBy;
+    this.distribution = resourceResponse._distribution;
+    this.rev = resourceResponse._rev;
+    this.deprecated = resourceResponse._deprecated;
+    this.data = Object.keys(resourceResponse)
```

Comment:
based on the [comment](https://github.com/BlueBrain/nexus-webapp-commons/pull/26/files#diff-5b4cca8f25650ffe3a2eb5a75cf123aaR7) below, could this just be:

```typescript
Object.keys(resourceResponse).forEach(key => {
  if(!resourceMetadataKeys.includes(key)) {
    this.data[key] = resourceResponse[key]
  }
})
```